### PR TITLE
Add default properties to link buttons and remove redundant settings

### PR DESCRIPTION
**Pull request by luixxiul**
Like a previous PR on link buttons, this PR sets additional properties to them as default ones. There should be no major regressions as I checked manually and set properties to elements which need specific settings.

This PR also sets `display: inline` to `link_inline` to reflect its name and cancel `inline-flex` inherited from `mx_AccessibleButton_hasKind`. `inline-flex` places button texts weirdly and breaks the design when they are over multiple lines.

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

Notes: none

type: task

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://pr8151--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/structures/auth/_Login.scss [hunk 1]
```diff
@@ -88,13 +88,12 @@ limitations under the License.
 div.mx_AccessibleButton_kind_link.mx_Login_forgot {
     display: block;
     margin: 0 auto;
```
Comment (luixxiul) [inline res/css/structures/auth/_Login.scss:90]: These will be able to be removed by creating a wrapper and setting `kind="link_inline"` in favor of `link`.

File: res/css/views/dialogs/_SpaceSettingsDialog.scss [hunk 2]
```diff
@@ -77,6 +77,7 @@ limitations under the License.
 
     .mx_AccessibleButton_hasKind {
         &.mx_AccessibleButton_kind_link {
+            font-size: $font-14px;
```
Comment (luixxiul) [inline res/css/views/dialogs/_SpaceSettingsDialog.scss:80]: after2

File: res/css/views/spaces/_SpaceBasicSettings.scss [hunk 3]
```diff
@@ -65,6 +65,7 @@ limitations under the License.
             display: inline-block;
             margin: auto 18px;
             color: #368bd6;
+            font-size: $font-14px; // See _SpaceSettingsDialog.scss
```
Comment (luixxiul) [inline res/css/views/spaces/_SpaceBasicSettings.scss:68]: after5

File: res/css/views/spaces/_SpaceCreateMenu.scss [hunk 4]
```diff
@@ -93,10 +93,6 @@ $spacePanelWidth: 68px;
             width: min-content;
         }
 
-        .mx_AccessibleButton_kind_link {
-            font-size: inherit;
```
Comment (luixxiul) [inline res/css/views/spaces/_SpaceCreateMenu.scss:97]: after

File: res/css/views/spaces/_SpaceCreateMenu.scss [hunk 5]
```diff
@@ -118,7 +114,5 @@ $spacePanelWidth: 68px;
     .mx_AccessibleButton_kind_link {
         color: $accent;
         position: relative;
-        font-size: inherit;
-        line-height: inherit;
```
Comment (luixxiul) [inline res/css/views/spaces/_SpaceCreateMenu.scss:122]: after

File: res/css/views/rooms/_PinnedEventTile.scss [hunk 6]
```diff
@@ -92,8 +92,6 @@ limitations under the License.
 
         .mx_AccessibleButton_kind_link {
             margin-left: 12px;
-            font-size: inherit;
-            line-height: inherit;
```
Comment (luixxiul) [inline res/css/views/rooms/_PinnedEventTile.scss:96]: pinned

File: res/css/views/rooms/_NewRoomIntro.scss [hunk 7]
```diff
@@ -23,10 +23,6 @@ limitations under the License.
         }
     }
 
-    .mx_AccessibleButton_kind_link {
-        font-size: inherit;
-    }
-
```
Comment (luixxiul) [inline res/css/views/rooms/_NewRoomIntro.scss:29]: newroom

File: res/css/structures/_UserMenu.scss [hunk 8]
```diff
@@ -147,11 +147,6 @@ limitations under the License.
                     margin-top: 8px;
                 }
             }
-
-            .mx_AccessibleButton_kind_link {
-                font-weight: normal;
-                font-size: inherit;
-            }
```
Comment (luixxiul) [inline res/css/structures/_UserMenu.scss:154]: guest

File: res/css/views/dialogs/_FeedbackDialog.scss [hunk 9]
```diff
@@ -58,10 +58,6 @@ limitations under the License.
             line-height: $font-15px;
         }
 
-        .mx_AccessibleButton_kind_link {
-            font-size: inherit;
-        }
-
```
Comment (luixxiul) [inline res/css/views/dialogs/_FeedbackDialog.scss:64]: feedback

File: res/css/views/dialogs/_GenericFeatureFeedbackDialog.scss [hunk 10]
```diff
@@ -21,9 +21,4 @@ limitations under the License.
         line-height: $font-20px;
         margin-bottom: 24px;
     }
-
-    .mx_AccessibleButton_kind_link {
-        font-size: inherit;
-        line-height: inherit;
-    }
```
Comment (luixxiul) [inline res/css/views/dialogs/_GenericFeatureFeedbackDialog.scss:28]: generic

File: res/css/views/dialogs/_SpotlightDialog.scss [hunk 11]
```diff
@@ -241,7 +241,6 @@ limitations under the License.
         .mx_SpotlightDialog_recentSearches > h4 > .mx_AccessibleButton_kind_link {
             padding: 0;
             float: right;
-            font-weight: normal;
```
Comment (luixxiul) [inline res/css/views/dialogs/_SpotlightDialog.scss:244]: spot

File: res/css/views/settings/tabs/room/_NotificationSettingsTab.scss [hunk 12]
```diff
@@ -50,10 +50,6 @@ limitations under the License.
             font-size: $font-12px;
             line-height: $font-15px;
             margin-right: 32px;
-
-            .mx_AccessibleButton_kind_link {
-                font-size: inherit;
-            }
```
Comment (luixxiul) [inline res/css/views/settings/tabs/room/_NotificationSettingsTab.scss:56]: notification

File: res/css/views/settings/_ProfileSettings.scss [hunk 13]
```diff
@@ -61,6 +61,7 @@ limitations under the License.
     margin-bottom: 28px;
 
     > .mx_AccessibleButton_kind_link {
+        font-size: $font-14px;
```
Comment (luixxiul) [inline res/css/views/settings/_ProfileSettings.scss:64]: profile

File: res/css/views/settings/tabs/user/_PreferencesUserSettingsTab.scss [hunk 14]
```diff
@@ -32,9 +32,5 @@ limitations under the License.
                 margin-top: 20px;
             }
         }
-
-        .mx_AccessibleButton_kind_link {
-            font-size: inherit;
-        }
```
Comment (luixxiul) [inline res/css/views/settings/tabs/user/_PreferencesUserSettingsTab.scss:38]: preference

File: res/css/structures/auth/_Login.scss [hunk 15]
```diff
@@ -88,13 +88,12 @@ limitations under the License.
...
-    // style it as a link
-    font-size: inherit;
```
Comment (luixxiul) [inline res/css/structures/auth/_Login.scss:92]: login

**Review by turt2live — APPROVED** [review]
I think this is generally fine, though as a precaution I'd like to land it after we cut the RC so we have an opportunity to fix any regressions on a more relaxed timescale - button code is historically notorious for causing issues, even before recent changes.

**Comment by luixxiul** [comment]
No problem about blocking until the next RC cut.

> button code is historically notorious for causing issues, even before recent changes.

I would agree. As far as I checked, the code is quite complex and vulnerable to regressions.

**Comment by luixxiul** [comment]
Thanks for checking- I'll keep an eye on a regression

**Pull request merged**